Gems/AtomLyIntegration/CommonFeatures/Code/Source/CoreLights/EditorAreaLightComponent.cpp

```diff
@@ -263,7 +263,23 @@ namespace AZ
 
             // Update the cached light type.
             m_lightType = m_controller.m_configuration.m_lightType;
+            
+            // Check to see if the current photometric type is supported by the light type. If not, convert to lumens before deactivating.
+            auto supportedPhotometricUnits = m_controller.m_configuration.GetValidPhotometricUnits();
+            auto foundIt = AZStd::find_if(
+                supportedPhotometricUnits.begin(),
+                supportedPhotometricUnits.end(),
+                [&](const Edit::EnumConstant<PhotometricUnit>& entry) -> bool
+                {
+                    return AZStd::RemoveEnum<PhotometricUnit>::type(m_controller.m_configuration.m_intensityMode) == entry.m_value;
+                }
+            );
 
+            if (foundIt == supportedPhotometricUnits.end())
```

Comment:
I wonder if it makes sense to wrap this part to a function of m_configuration.

Reply:
At least for now this is a pretty editor specific problem, and requires the correct timing - ConvertToIntensityMode will only work when the component is active. This whole function is a bit janky, but its working within the limits of the editor component. Ultimately I'd like it to be possible to modify the light type and do all the work to automatically create the correct shape through the behavior context on the controller, but I don't think that's possible with the current setup.

Reply:
At least for now this is a pretty editor specific problem, and requires the correct timing - ConvertToIntensityMode will only work when the component is active. This whole function is a bit janky, but its working within the limits of the editor component. Ultimately I'd like it to be possible to modify the light type and do all the work to automatically create the correct shape through the behavior context on the controller, but I don't think that's possible with the current setup.